### ISSUE TITLE
Draft_17 Refuses Handshake

### ISSUE THREAD

**Issue by Librichard** [TooTallNate/Java-WebSocket#759]
## Expected Behavior
Connected to WSS server

## Current Behavior
	 Server: /xxx:443
	 Code: -1
	 Reason: draft org.java_websocket.drafts.Draft_17@46c04b76 refuses handshake
	 Remote: false

## Possible Solution
If we use Jetty WebSocket implementation, it works.

## Debug log (for bugs)
write(157): {GET /xfer HTTP/1.1
Connection: Upgrade
Host: xxx:443
Sec-WebSocket-Key: xxx
Sec-WebSocket-Version: 13
Upgrade: websocket

}
- I removed actual IP and key value
- I am suspicious that a line like: "Sec-WebSocket-Protocol: xfer" is missing

## Context
Traffic Control Systems

## Your Environment

* Version used: java-websocket-1.3.3
* Java version: 1.8
* Operating System and version: Windows 10 Pro 64bit
* Endpoint Name and version: It is a Siemens Server

**Comment by marci4** [comment]
Hello @Librichard,
Please update to the latest version of this library 1.3.9!

Draft_17 got removed a long time ago. The only supported draft is now Draft_6455.

Best regards,
marci4

**Comment by Librichard** [comment]
I did, and now I got this error:
	 Server: /xxx:443
	 Code: 1002
	 Reason: Invalid status code received: 400 Status line: HTTP/1.1 400 Bad Request
	 Remote: false

**Comment by marci4** [comment]
Hallo @Librichard,

could you maybe tell my what you are trying to achieve....

A few things I can see:

- There is no Sec-WebSocket-Protocol extension registered with the name `xfer` (See [this](https://www.iana.org/assignments/websocket/websocket.xml)). Are you sure that this extension does really exist?
- 400 is the response by your server. This is not a valid http code, which has to be 101.... So you maybe need to check if you are requesting the correct endpoint.
- It also looks as your server does not support the Sec-WebSocket-Protocol `xfer` since it is not part of the handshake response.

Best regards,
marci4

**Comment by Librichard** [comment]
@marci4 ,
xfer is a protocol on top of WebSocket for a traffic control system. It is implemented in the servers I am dealing with by Siemens. It works fine when I use the Jetty implementation of WebSockets, but it fails when I use yours. Since we are using your jar library in our systems already (java-websockets-v1.3.3), I rather to keep using it, otherwise we have to add the Jetty library to our system.

**Comment by marci4** [comment]
Hello @Librichard,
I was not able to find anything on this Sec-WebSocket-Protocol on the internet. 
Could you maybe mail me some more information (if you want to keep it confidential).

Also could you maybe include a debug log using WebSocketImpl.DEBUG = TRUE as well as the way you connect using jetty and the way you try to connect using this library. 

Best regards,
marci4

**Comment by marci4** [comment]
Any update?

**Comment by Librichard** [comment]
I am not working on that project anymore. Actually I don't work there anymore.
I think you didn't find anything about XFER because it is proprietary. I set the DEBUG = TRUE, that's how I got all the client request details.
It seems jetty does not have the idea of drafts. Its API is different from yours. For some reason, the XFER server is able to handshake with our client using their library (following their API contract), but not yours.
